dataflows/processors/dumpers/file_formats.py

```diff
@@ -37,17 +38,21 @@ def prepare_resource(cls, resource):
 
     def __transform_row(self, row):
         try:
-            return dict((k, self.__transform_value(v, self.fields[k].type))
+            return dict((k, self.__transform_value(v, self.fields[k]))
                         for k, v in row.items())
         except Exception:
             logging.exception('Failed to transform row %r', row)
             raise
 
-    @classmethod
-    def __transform_value(cls, value, field_type):
+    def __transform_value(self, value, field):
         if value is None:
-            return cls.NULL_VALUE
-        serializer = cls.SERIALIZERS.get(field_type, cls.DEFAULT_SERIALIZER)
+            return self.NULL_VALUE
+        serializer = self.SERIALIZERS.get(field.type, self.DEFAULT_SERIALIZER)
+        if self.temporal_format_property:
```

Comment:
Performance wise, it would be better to run this check once in __init__ and not on every value transformation.

In the constructor I would deepcopy the SERIALIZERS (so that we don't change the class variable) and modify it once based on this logic.

Reply:
@akariv 
Thanks. Updated. Please take a look